components/procore/sources/budget-snapshot/budget-snapshot.js

```diff
@@ -0,0 +1,50 @@
+const common = require("../common.js");
+
+module.exports = {
+  ...common,
+  name: "New or Updated Budget Snapshot (Instant)",
+  key: "procore-budget-snapshot",
+  description:
+    "Emits an event each time a Budget Snapshot is created, updated, or deleted in a project.",
+  version: "0.0.1",
+  methods: {
+    ...common.methods,
+    getResourceName() {
+      return "Budget View Snapshots";
+    },
+    async getDataToEmit(body) {
+      const { resource_id: resourceId } = body;
+      const limit = 100;
+      let offset = 0;
+      let total = limit;
+      let snapshotRows = [];
+      while (total == limit) {
+        const resource = await this.procore.getBudgetViewSnapshot(
+          this.company,
+          this.project,
+          resourceId,
+          limit,
+          offset
+        );
+        snapshotRows = snapshotRows.concat(resource);
+        total = resource.length;
+        offset += limit;
+      }
+      return { body, snapshotRows };
```

Comment:
I wouldn't expect the event to query the whole object before emitting the event downstream.
Sending a resource ID is fine IMHO. cc @dylburger WDYT?